src/cookbooks/bigquery.md

```diff
@@ -190,6 +191,19 @@ You can create views in BigQuery if you have access via [GCP BigQuery Console](b
     - Prefix your view with your username. If your username is `username@mozilla.com` create a table with `username_my_view`.
 - See [Creating Views](https://cloud.google.com/bigquery/docs/views) documentation for detailed steps.
 
+### Accessing map-like fields
+
+BigQuery currently lacks native map support and our workaround is to use a STRUCT type with fields named [key, value]. We've created a user-defined function (UDFs) that provides key-based access with the signature: `udf.get_key(<struct>, <key>)`. The example below generates a count per `reason` key in the `event_map_values` field in the telemetry events table for Normandy unenrollment events from yesterday.
```

Comment:
There's a bit more nuance needed here, unfortunately. The `get_key` function is specific to the way we transform Parquet datasets to BigQuery, where we transform map types to a single-field struct that contains a repeated struct named `key_value` that contains key/value pairs. The `get_key` function expects a struct with a field literally named `key_value`. This works for datasets imported from Parquet, but not for maps unrolled by the GCP pipeline.

We should either include that nuance here or refactor get_key to be more general.

Reply:
See https://github.com/mozilla/bigquery-etl/pull/216 for the proposed refactor. It looks like we have no existing uses in bigquery-etl, so this change won't break any existing use cases.

Reply:
@relud is making an effort to get this resolved in the near future, so probably not worth mentioning for the time being.

Let's go ahead and merge the docs as-is, and then we will hopefully have a backend refactor in place by EOW that maintains the same interface described here.